cou/utils/juju_utils.py

```diff
@@ -131,6 +133,67 @@ async def wrapper(*args: Any, **kwargs: Any) -> Any:  # pylint: disable=W9011
     return _wrapper
 
 
+@dataclass(frozen=True)
+class COUMachine:
```

Comment:
nit: does the `COU` prefix mean much here? These classes seem quite generic

Reply:
No, it's just to differentiate from juju Machine, Application and Unit. I do not think that it's necessary, but I followed the same approach as we did with COUModel.